test.rkt

```diff
@@ -31,10 +31,19 @@
            (or (equal? pt (ival-lo ival))
                (equal? pt (ival-hi ival))))))
 
-(define (value-equals? bf1 bf2)
+(define (value-equals? bf1 bf2 [rnd-mode 'nearest])
   (if (boolean? bf1)
       (equal? bf1 bf2)
-      (or (bf= bf1 bf2) (and (bfnan? bf1) (bfnan? bf2)))))
+      (or (bigfloats-equal? bf1 bf2 rnd-mode) (and (bfnan? bf1) (bfnan? bf2)))))
+
+(define (bigfloats-equal? x y rnd-mode)
+  (cond
+     [(< (bigfloat-precision x) (bigfloat-precision y))
+      (bf= x (parameterize ([bf-rounding-mode rnd-mode] [bf-precision (bigfloat-precision x)]) (bfcopy y)))]
+     [(> (bigfloat-precision x) (bigfloat-precision y))
+      (bf= y (parameterize ([bf-rounding-mode rnd-mode] [bf-precision (bigfloat-precision y)]) (bfcopy x)))]
+     [else (bf= y x)]))
```

Comment:
This is fine but would be nicer to just round both `x` and `y` to the minimum precision, I don't think this code needs to be fast since it's test code.